### PR TITLE
refactor(apport): Replace global proc_pid_fd by ProcPid

### PR DESCRIPTION
**Pull request by bdrung**
Replace global variable `proc_pid_fd` by an `ProcPid` instance that is a context manager to ensure the opened file will be closed.

### CODE REVIEW

**Comment by bdrung** [comment]
IMO we can ignore codecov/project since the merge reduces the amount of code.

File: data/apport [hunk 1]
```diff
@@ -49,6 +49,22 @@ import apport.report
 LOG_FORMAT = "%(levelname)s: apport (pid %(process)s) %(asctime)s: %(message)s"
 
 
+class ProcPid:
+    """Access /proc/<pid>."""
+
+    def __init__(self, pid: int, path: typing.Optional[str] = None) -> None:
+        self.fd = os.open(
+            path or f"/proc/{pid}", os.O_RDONLY | os.O_PATH | os.O_DIRECTORY
+        )
```
Comment (schopin-pro) [inline data/apport:58]: issue (design): should we really embed a FD in a plain class?

This pattern means we're relying on the finalizer closing out the FD, which sounds like a bad idea to me.

**Review by schopin-pro — CHANGES_REQUESTED** [review]
I have a design issue that I'd like addressed.

File: data/apport (hunk 1 above)
Comment (bdrung) [reply data/apport:58]: Very good point. It will be probably much better to convert that class into a context manager.

**Review by schopin-pro — COMMENTED** [review]
question: doesn't that open us to race condition?

Disclaimer: I'm really not well-versed in the particulars of procfs, crash-handling, and all the low-level notions here, so I might be raising moot points.

In the previous design, it's fairly obvious that we only open the PID directory once, which I'd assume protects us in case of PID recycling by the kernel (can it recycle the PID of a process that just crashed?). Do we still have that "atomic" property with the new one? I *think* we do, but I can't figure out the call tree well enough to be sure.

**Comment by bdrung** [comment]
It will not open us to race conditions. We create only one `ProcPid` object for the relevant PID.

**Pull request merged**